packages/smart/contracts/turbo/AMMFactory.sol

```diff
@@ -45,33 +56,45 @@ contract AMMFactory is BNum {
         uint256 price
     );
 
-    constructor(BFactory _bFactory, uint256 _fee) {
+    constructor(
+        BFactory _bFactory,
+        MasterChef _masterChef,
+        uint256 _fee
+    ) {
         bFactory = _bFactory;
+        masterChef = _masterChef;
         fee = _fee;
     }
 
+    function approveMarketFactory(address _marketFactory) external onlyOwner returns (bool) {
+        approvedMarketFactories[_marketFactory] = true;
+        return true;
+    }
+
     function createPool(
         AbstractMarketFactory _marketFactory,
         uint256 _marketId,
         uint256 _initialLiquidity,
         uint256[] memory _weights,
         address _lpTokenRecipient
     ) public returns (uint256) {
+        require(approvedMarketFactories[address(_marketFactory)], "MarketFactory not approved to create pool.");
         require(pools[address(_marketFactory)][_marketId] == BPool(0), "Pool already created");
 
         AbstractMarketFactory.Market memory _market = _marketFactory.getMarket(_marketId);
 
         require(_weights.length == _market.shareTokens.length, "Must have one weight for each share token");
 
+        uint256 _sets = _marketFactory.calcShares(_initialLiquidity);
+        require(_sets >= MIN_INITIAL_LIQUIDITY, "Initial liquidity must be at least 100 collateral.");
```

Comment:
Is the value here accurate? MIN_INITIAL_LIQUIDITY uses BONE which afaik is 1e18. But USDC and such are based on 1e6.

Reply:
The check is done with shares because it is normalized to 10e18. Will add a comment.